src/compiler/checker.ts

```diff
@@ -15251,6 +15296,23 @@ namespace ts {
             return str;
         }
 
+        function applyNumberMapping(symbol: Symbol, n: NumberLiteralType): Type {
+            switch (intrinsicTypeKinds.get(symbol.escapedName as string)) {
+                case IntrinsicTypeKind.Integer: return getNumberLiteralType(Math.floor(n.value));
+            }
+            return n;
+        }
+
+        function applyNumberMapping2(symbol: Symbol, a: NumberLiteralType, b: NumberLiteralType): Type {

```

Comment:
There is quite some code duplication, what do you think about for instance:

```js
const MathArity2: Record<string, (a: number, b: number) => number> = {
    Add  : (a, b) => a +  b,
    Sub  : (a, b) => a -  b,
    Mul  : (a, b) => a *  b,
    Div  : (a, b) => a /  b,
    Pow  : (a, b) => a ** b,
    Hypot: (a, b) => Math.hypot(a, b),
    Min  : (a, b) => Math.min  (a, b),
    Max  : (a, b) => Math.max  (a, b),
};
```

And `MathArity1` could have `exp` (`E ** x`), `sin`, `cos`, `tan`, `abs`, `round`, `sqrt` and your `Integer` (or just `floor` to keep it in sync with JS?).

(`MathArity0` would be simply constants like `PI` and `E`, probably just a constants in the `d.ts`?)